pkg/cluster/cluster.go

```diff
@@ -229,6 +229,9 @@ func (c *Cluster) run(stopC <-chan struct{}, wg *sync.WaitGroup) {
 		case event := <-c.eventCh:
 			switch event.typ {
 			case eventModifyCluster:
+				if isSpecEqual(event.cluster.Spec, c.cluster.Spec) {
```

Comment:
We might have more lines of code logic to process, e.g. handling upgrade in progress. Better not put them in braces.